cou/steps/execute.py

```diff
@@ -89,25 +88,45 @@ async def _run_step(step: UpgradeStep, interactive: bool) -> None:
             await apply_plan(sub_step, interactive)
 
 
-async def apply_plan(plan: UpgradeStep, interactive: bool) -> None:
+async def apply_plan(plan: BaseStep, interactive: bool) -> None:
```

Comment:
Can we rename it to:
```suggestion
async def apply_step(step: BaseStep, interactive: bool) -> None:
```